samples/bluetooth/iso_broadcast_benchmark/src/receiver.c

```diff
@@ -0,0 +1,510 @@
+	 * the lowest 2 digits
+	 */
+	uint8_t integer;
+	uint8_t fraction;
+	uint32_t total_packets;
+	uint64_t tmp;
+
+	total_packets = stats->iso_recv_count + stats->iso_lost_count;
+	tmp = ((uint64_t)stats->iso_recv_count * 10000) / total_packets;
+
+	integer = tmp / 100;
+	fraction = tmp - integer * 100;
+
+	LOG_INF("%s: Received %u/%u (%u.%02u%%) - Total packets lost %u",
+		name, stats->iso_recv_count, total_packets, integer, fraction,
+		stats->iso_lost_count);
+
+}
+
+static void iso_recv(struct bt_iso_chan *chan, struct net_buf *buf)
+{
+	uint32_t counter = sys_get_le32(buf->data);
+	uint32_t total_packets;
+	static bool latest_stats_arr[1000];
+	static size_t latest_stats_arr_pos;
+	uint32_t packets_lost;
+
+	/* NOTE: The packets received may be on different BISes */
+
+	if (buf->len == 0) {
+		LOG_INF("Data length is zero - Bad packet");
+		return;
+	}
+
+	if (counter <= last_received_counter) {
+		LOG_INF("Counter (%u) was less or to last counter received (%u)",
+			counter, last_received_counter);
+		return;
+	}
+
+	packets_lost = counter - last_received_counter - 1;
+
+	if (packets_lost > 0 && last_received_counter > 0) {
+
+		current_sync_stats.iso_lost_count += packets_lost;
+		overall_stats.iso_lost_count += packets_lost;
+
+		for (int i = 0; i < packets_lost; i++) {
+			latest_stats_arr[latest_stats_arr_pos++] = false;
+			if (latest_stats_arr_pos == sizeof(latest_stats_arr)) {
+				latest_stats_arr_pos = 0;
+			}
+		}
+	}
+
+	current_sync_stats.iso_recv_count++;
+	overall_stats.iso_recv_count++;
+
+	latest_stats_arr[latest_stats_arr_pos++] = true;
+	if (latest_stats_arr_pos == sizeof(latest_stats_arr)) {
+		latest_stats_arr_pos = 0;
+	}
+
+	last_received_counter = counter;
+
+	total_packets = overall_stats.iso_recv_count + overall_stats.iso_lost_count;
+
+	if ((total_packets % 100) == 0) {
+		struct iso_recv_stats latest_stats = { 0 };
+
+		for (int i = 0; i < ARRAY_SIZE(latest_stats_arr); i++) {
+			if (i == total_packets) {
+				break;
+			}
+
+			if (latest_stats_arr[i]) {
+				latest_stats.iso_recv_count++;
+			} else {
+				latest_stats.iso_lost_count++;
+			}
+		}
+
+		print_stats("Overall     ", &overall_stats);
+		print_stats("Current Sync", &current_sync_stats);
+		print_stats("Latest 1000 ", &latest_stats);
+	}
+}
+
+static void iso_connected(struct bt_iso_chan *chan)
+{
+	LOG_INF("ISO Channel %p connected", chan);
+
+	big_sync_start_time = k_uptime_get();
+
+	k_sem_give(&sem_big_sync);
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	/* Calculate cumulative moving average - Be aware that this may
+	 * cause overflow at sufficiently large counts or durations
+	 */
+	static int64_t average_duration;
+	int64_t sync_duration = k_uptime_delta(&big_sync_start_time);
+	int64_t total_duration = sync_duration + (big_sync_count - 1) * average_duration;
+
+	average_duration = total_duration / big_sync_count;
+
+	LOG_INF("ISO Channel %p disconnected with reason 0x%02x after "
```

Comment:
I got a disconnect after a few minutes with 356044198986706 ms, so there is something fishy here.

Reply:
Right - Completely forgot to update this - Found severe issues with the timing on the broadcaster side. Will find a solution for it :)

Reply:
Seems the update to the logging also fixes this :)